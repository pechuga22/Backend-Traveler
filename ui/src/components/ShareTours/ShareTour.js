import React, { useState, useEffect, useRef } from 'react'
import { Button, Header } from 'semantic-ui-react'
import { CopyToClipboard } from 'react-copy-to-clipboard'

import './ShareTour.less'

export default function ShareTour() {
  const uri = window.location.href
  const [copied, setCopied] = useState(false)
  const [url, setUrl] = useState(uri)
  const timeout = useRef()

  function onCopyUrl() {
    if (!copied) {
      setUrl(uri)
      setCopied(true)
      timeout.current = setTimeout(() => setCopied(false), 1000)
    }
  }

  useEffect(() => {
    if (timeout.current) {
      clearTimeout(timeout.current)
    }
  }, [])

  return (
    <div>
      <CopyToClipboard text={url} onCopy={onCopyUrl}>
        <Button size='big' icon='share alternate' className='btnshare' />
      </CopyToClipboard>
      {copied ? (
        <Header as='h5' ref={timeout} className='copied' color='pink'>
          copied!
        </Header>
      ) : null}
    </div>
  )
}
