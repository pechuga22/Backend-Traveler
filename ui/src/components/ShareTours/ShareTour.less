.copied {
  right: 2%;
  top: 45px;
  z-index: 10;
  position: fixed;
}
