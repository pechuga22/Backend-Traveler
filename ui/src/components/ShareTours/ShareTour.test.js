import React from 'react'
import { shallow} from 'enzyme'
import toJson from 'enzyme-to-json'
import { render } from '@testing-library/react'

import ShareTour from './ShareTour'

describe('./components/ShareTour/ShareTour.js', () => {
  describe('.render()', () => {
    it('should render ShareTour component', () => {
      const wrapper = shallow(<ShareTour />)
      expect(toJson(wrapper)).toMatchSnapshot()
    })

    it('renders text ShareTour', () => {
      const wrapper = shallow(<ShareTour />)
      const copied = wrapper.contains('copied').valueOf(true)
      if (copied) {
        const { getByText } = render(<ShareTour />)
        const loadingText = getByText(/copied!/i)
      }
    })
  })
})