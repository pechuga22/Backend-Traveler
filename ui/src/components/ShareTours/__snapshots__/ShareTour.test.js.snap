// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`./components/ShareTour/ShareTour.js .render() should render ShareTour component 1`] = `
<div>
  <CopyToClipboard
    onCopy={[Function]}
    text="http://localhost/"
  >
    <Button
      as="button"
      className="btnshare"
      icon="share alternate"
      size="big"
    />
  </CopyToClipboard>
</div>
`;
